beacon/src/beacon.rs

```diff
@@ -0,0 +1,84 @@
+use crate::Entropy;
+use helium_proto::{
+    services::poc_lora,
+    {BlockchainRegionParamV1, DataRate},
+};
+use rand::{seq::SliceRandom, Rng, SeedableRng};
+use sha2::{Digest, Sha256};
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+pub const BEACON_PAYLOAD_SIZE: usize = 10;
+
+#[derive(Debug, Clone)]
+pub struct Beacon {
+    pub data: Vec<u8>,
+
+    pub frequency: u64,
+    pub datarate: DataRate,
+    pub remote_entropy: Entropy,
+    pub local_entropy: Entropy,
+}
+
+impl Beacon {
+    pub fn new(
+        remote_entropy: Entropy,
+        local_entropy: Entropy,
+        region_params: &[BlockchainRegionParamV1],
+    ) -> Self {
+        let data = {
+            let mut hasher = Sha256::new();
+            remote_entropy.digest(&mut hasher);
+            local_entropy.digest(&mut hasher);
+            hasher.finalize().to_vec()
+        };
+        let mut seed = [0u8; 32];
+        seed.copy_from_slice(&data[0..32]);
+        let mut rng = rand_chacha::ChaCha12Rng::from_seed(seed);
+
+        let frequency = rand_frequency(region_params, &mut rng);
+        let datarate = DataRate::Sf7bw125;
+
+        Self {
+            data: data[0..BEACON_PAYLOAD_SIZE].to_vec(),
```

Comment:
You already have a Vec that's about to drop, might as well use it
```suggestion
            data: {
                data.truncate(BEACON_PAYLOAD_SIZE);
                data
            },
```